### PR TITLE
Add an env goal

### PR DESCRIPTION
**Pull request by chonton**
Use 'docker-machine env' to set the java process environment variables.  This allows less environmental setup for jenkins.

### CODE REVIEW

**Comment by rhuss** [comment]
thanks will check this out for the next release (which will come sooner than the last one)

**Comment by chonton** [comment]
We will also want to add to Docker lifecycle.  Probably bind to _validate_ or _initialize_ phase.

File: src/main/java/io/fabric8/maven/docker/MachineMojo.java [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+        try {
+            outputStream.close();
+        } catch (IOException e) {
+            log.info("failed to close docker-machine output stream: " + e.getMessage());
+        }
+    }
+
+    private Process startDockerMachineProcess() throws MojoExecutionException {
+        try {
+            return Runtime.getRuntime().exec(new String[] { "docker-machine", "env", "--shell", "cmd" });
```
Comment (rhuss) [inline src/main/java/io/fabric8/maven/docker/MachineMojo.java]: This wont work if you have more than one docker machine installed.

**Comment by rhuss** [comment]
I wonder whether we really need a new goal (which is always something 'heavy weight'), but maybe a proeprty instead which gets evaluated in the base Mojo for adding to the properties. Or even, as an automatic fallback when not DOCKER_HOST is set ? This would make things much more transparent for people.

**Comment by rhuss** [comment]
Maybe we can add it the evaluation of docker-machine as fallback here --> https://github.com/fabric8io/docker-maven-plugin/blob/master/src/main/java/io/fabric8/maven/docker/util/EnvUtil.java#L25-L33 (should be checked whether docker-machine is installed first, though)

File: src/main/java/io/fabric8/maven/docker/MachineMojo.java (hunk 1 above)
Comment (chonton) [inline src/main/java/io/fabric8/maven/docker/MachineMojo.java]: Warn and choose one at random or fail?

**Comment by chonton** [comment]
I like the idea of this being a fallback mechanism.  I'll look at this tonight or tomorrow.

**Comment by chonton** [comment]
Updating the pull request per your comments.

**Pull request closed without merging**